src/interpreter/InterpreterState.ts

```diff
@@ -4,9 +4,14 @@
  */
 
 import {SimpleCellAddress} from '../Cell'
+import {FormulaVertex} from '../DependencyGraph/FormulaCellVertex'
 
 export class InterpreterState {
-  constructor(public formulaAddress: SimpleCellAddress, public arraysFlag: boolean) {
+  constructor(
+    public formulaAddress: SimpleCellAddress,
+    public arraysFlag: boolean,
+    public formulaVertex?: FormulaVertex,
```

Comment:
in what scenarios formulaVertex can be not set?

Reply:
[`calculateFormula`](https://github.com/handsontable/hyperformula/blob/master/src/HyperFormula.ts#L3529)

which runs `runAndForget`, ast evaluation without creating vertex.

Reply:
right!